### PR TITLE
Update multiple docs pages to follow Docs Content style guide

### PR DESCRIPTION
**Pull request by amandeepmittal**
This PR addresses minor docs-related styling issues on the Getting Started, Data Fetching (Get Server Side Props), etc pages. It follows the mechanics of the Docs Content Style guide to maintain consistency across all documentation. 

Also fixes some minor issues such as missing period at the end of a sentence on docs pages like as Data Fetching (getStaticProps).

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

**Review by molebox — DISMISSED** [review]

File: docs/basic-features/data-fetching/get-static-props.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ When a page with `getStaticProps` is pre-rendered at build time, in addition to
 
 This JSON file will be used in client-side routing through [`next/link`](/docs/api-reference/next/link.md) or [`next/router`](/docs/api-reference/next/router.md). When you navigate to a page that’s pre-rendered using `getStaticProps`, Next.js fetches this JSON file (pre-computed at build time) and uses it as the props for the page component. This means that client-side page transitions will **not** call `getStaticProps` as only the exported JSON is used.
 
-When using Incremental Static Generation `getStaticProps` will be executed out of band to generate the JSON needed for client-side navigation. You may see this in the form of multiple requests being made for the same page, however, this is intended and has no impact on end-user performance
+When using Incremental Static Generation `getStaticProps` will be executed out of band to generate the JSON needed for client-side navigation. You may see this in the form of multiple requests being made for the same page, however, this is intended and has no impact on end-user performance.
```
Comment (leerob) [inline docs/basic-features/data-fetching/get-static-props.md:87]:
```suggestion
When using Incremental Static Generation, `getStaticProps` will be executed "out of band" to generate the JSON needed for client-side navigation. You may see this in the form of multiple requests being made for the same page, however, this is intended and has no impact on end-user performance.
```

Comment (leerob) [reply docs/basic-features/data-fetching/get-static-props.md:87]: I feel like we need to define what out of band means here - it's not entirely clear.

Your changes look great though 😄

Comment (amandeepmittal) [reply docs/basic-features/data-fetching/get-static-props.md]: Thanks Lee! 

I am not sure what does that means in this context either. Do you have more context that you can share with me?

Comment (amandeepmittal) [reply docs/basic-features/data-fetching/get-static-props.md]: @leerob can you please take a look if the above rephrasing is accurate or not?

Comment (leerob) [reply docs/basic-features/data-fetching/get-static-props.md]: I like it!

Comment (amandeepmittal) [reply docs/basic-features/data-fetching/get-static-props.md]: Awesome. Thanks!

**Review by leerob — APPROVED** [review]

**Pull request merged**